include/votca/xtp/aomatrix.h

```diff
@@ -31,265 +31,313 @@
-        const ub::matrix<double> &getNuclearpotential()const{ return _nuclearpotential;}
-        ub::matrix<double> &getExternalpotential(){ return _externalpotential;}
-        const ub::matrix<double> &getExternalpotential()const{ return _externalpotential;}
-        // ~AOESP();
-    private:    
-        ub::matrix<double> _nuclearpotential;
-        ub::matrix<double> _externalpotential;
-    };
-    
-    
-    
-    // derived class for Effective Core Potentials
-    class AOECP : public AOMatrix{
-    public:
-        //block fill for overlap, implementation in aoesp.cc
-        void FillBlock( ub::matrix_range< ub::matrix<double> >& _matrix,const AOShell* _shell_row,const AOShell* _shell_col, AOBasis* ecp);
-
-        
-        typedef boost::multi_array<double, 3> type_3D;
-        
-        
-        ub::matrix<double> calcVNLmatrix(int _lmax_ecp,const vec& posC, const AOGaussianPrimitive& _g_row,const AOGaussianPrimitive& _g_col,const  ub::matrix<int>& _power_ecp,const ub::matrix<double>& _gamma_ecp,const ub::matrix<double>& _pref_ecp   );
-        
-        
-        
-        void getBLMCOF(int _lmax_ecp, int _lmax_dft, const vec& pos, type_3D& BLC, type_3D& C  );
-        ub::vector<double> CalcNorms( double decay,int size);
-        ub::vector<double> CalcInt_r_exp( int nmax, double decay );
-    };
-    
-
-    
-    // derived class for kinetic energy
-    class AOKinetic : public AOMatrix{
-    public:
-        //block fill for overlap, implementation in aokinetic.cc
-        void FillBlock( ub::matrix_range< ub::matrix<double> >& _matrix, const AOShell* _shell_row, const AOShell* _shell_col , AOBasis* ecp);
- 
-        
-    };
-    
-    
-    // derived class for atomic orbital overlap
-    class AOOverlap : public AOMatrix{
-    public:
-        //block fill for overlap, implementation in aooverlap.cc
-        void FillBlock( ub::matrix_range< ub::matrix<double> >& _matrix,const AOShell* _shell_row,const AOShell* _shell_col, AOBasis* ecp);
-        //void Print();
-        
-	//        ~AOOverlap();
-        
-    };
-    
-    class AODipole_Potential : public AOMatrix{
-    public:
-        //block fill for overlap, implementation in aooverlap.cc
-        void FillBlock( ub::matrix_range< ub::matrix<double> >& _matrix,const AOShell* _shell_row,const AOShell* _shell_col, AOBasis* ecp);
-        
-        void Fillextpotential(const AOBasis& aobasis, const std::vector<ctp::PolarSeg*>& _sites);
-        ub::matrix<double> &getExternalpotential(){ return _externalpotential;}
-        const ub::matrix<double> &getExternalpotential()const{ return _externalpotential;}
-        
-        //void Print();
-        
-	//        ~AOOverlap();
-    private: 
-        void setAPolarSite(ctp::APolarSite* site){
-            apolarsite=site;
+}
+
+namespace votca {
+    namespace xtp {
+        namespace ub = boost::numeric::ublas;
+
+        /* "superclass" AOSuperMatrix contains all common functionality for
```

Comment:
It would be preferable to reserve the /* comment type for debugging and use // for comments in the code.

Reply:
What is this branch for exactly?

Reply:
1) agreed 
2) Due to some merges, the true purpose of this branch got a little mixed, it mainly focuses on electronic coupling using a linker atom